src/device.cpp

```diff
@@ -515,6 +518,110 @@ bool cvk_device::init_time_management(VkInstance instance) {
     return true;
 }
 
+// Returns the pipeline cache file path for a given Vulkan implementation.
+// If pipeline cache serialization is not enabled, an empty string is returned.
+std::string get_pipeline_cache_filename(VkPhysicalDeviceProperties properties) {
+    const char* cache_dir = getenv("CLVK_CACHE_DIR");
+    if (cache_dir == nullptr) {
+        return "";
+    }
+
+    // The pipeline cache file path is:
+    // ${CLVK_CACHE_DIR}/<UUID>.clvk-pipeline-cache.bin
```

Comment:
Minor detail: maybe the fixed part could come before the UUID so all pipeline cache files are grouped together when listing the content of the cache directory (i.e. `clvk-pipeline-cache.<UUID>.bin`), unless you were thinking of reusing the UUID to prefix cached SPIR-V files as well. I don't really mind either way.